### PR TITLE
Adds support of named ports for kubectl port-forward

### LINKED ISSUE #538: port-forward should support named ports
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):

FEATURE REQUEST

**Kubernetes version** (use `kubectl version`): v1.11.3

**Environment**:
- **Cloud provider or hardware configuration**: Libvirt
- **OS** (e.g. from /etc/os-release): Client on macOS 10.14.0, server CentOS 7
- **Kernel** (e.g. `uname -a`): 3.10.0-862.9.1.el7.x86_64
- **Install tools**: kubeadm
- **Others**:

**What happened**:

```
$ kubectl port-forward http-server http
error: Error parsing local port 'http': strconv.ParseUint: parsing "http": invalid syntax
```

**What you expected to happen**:

```
$ kubectl port-forward http-server http
Forwarding from 127.0.0.1:8080 -> 8080
Forwarding from [::1]:8080 -> 8080
```

**How to reproduce it** (as minimally and precisely as possible):

pod.yaml:
```
apiVersion: v1
kind: Pod
metadata:
  name: http-server
spec:
  containers:
  - image: gcr.io/google_containers/defaultbackend:1.4
    imagePullPolicy: IfNotPresent
    name: default-http-backend
    ports:
    - containerPort: 8080
      name: http
      protocol: TCP
```

```
kubectl create -f pod.yaml
kubectl port-forward  http-server web
```

### PR DESCRIPTION
**Pull request by m1kola**
**What this PR does / why we need it**:

Adds support of named ports for `kubectl port-forward`, so users will be able to run commands like this:

```
kubectl port-forward http-server http
kubectl port-forward http-server :http
kubectl port-forward http-server 8000:http
```

Same will work with services (including translation of the service's port into the pod's port).

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubernetes/kubectl/issues/538

**Special notes for your reviewer**:

Please, note that this PR also includes #69403 which needs to be reviewed and merged first (it makes an util function to be exportable, so we can use it in the port-forward command).

**Release note**:

```release-note
It is now possible to use named ports in the `kubectl port-forward` command
```

/sig cli
/kind feature

### CODE REVIEW

File: pkg/kubectl/cmd/portforward.go [hunk 1]
```diff
@@ -138,38 +138,81 @@ func (f *defaultPortForwarder) ForwardPorts(method string, url *url.URL, opts Po
...
 
+// splitPort splits port string which is in form of [LOCAL PORT]:REMOTE PORT
+// and returns local and remote ports separately
+func splitPort(port string) (local, remote string) {
+	parts := strings.Split(port, ":")
+	if len(parts) == 2 {
+		return parts[0], parts[1]
+	}
+
+	return parts[0], parts[0]
```
Comment (juanvallejo) [inline pkg/kubectl/cmd/portforward.go]: nit: just return "" for the second value here

File: pkg/kubectl/cmd/portforward.go [hunk 2]
```diff
@@ -138,38 +138,81 @@ func (f *defaultPortForwarder) ForwardPorts(method string, url *url.URL, opts Po
...
+		portnum, err := strconv.Atoi(remotePort)
 		if err != nil {
-			return ports, err
+			svcPort, err := util.LookupServicePortNumberByName(svc, remotePort)
+			if err != nil {
+				return nil, err
+			}
+			portnum = int(svcPort)
+
+			if localPort == remotePort {
```
Comment (juanvallejo) [inline pkg/kubectl/cmd/portforward.go]: after addressing the above comment:

```
if len(removePort) == 0
```

**Comment by m1kola** [comment]
Rebased

File: pkg/kubectl/cmd/portforward/portforward.go [hunk 3]
```diff
@@ -138,38 +138,81 @@ func (f *defaultPortForwarder) ForwardPorts(method string, url *url.URL, opts Po
...
+	var converted []string
+	for _, port := range ports {
+		localPort, remotePort := splitPort(port)
+
+		containerPortStr := remotePort
+		_, err := strconv.Atoi(remotePort)
+		if err != nil {
+			containerPort, err := util.LookupContainerPortNumberByName(pod, remotePort)
+			if err != nil {
+				return ports, err
```
Comment (juanvallejo) [inline pkg/kubectl/cmd/portforward/portforward.go]: nit, just return `nil, err` here

Comment (m1kola) [reply pkg/kubectl/cmd/portforward/portforward.go]: Thanks, updated.

**Comment by juanvallejo** [comment]
@m1kola thanks for the PR.
cc @soltysh for approval

/lgtm

**Comment by juanvallejo** [comment]
@m1kola please squash into single commit

**Comment by m1kola** [comment]
@juanvallejo please, re-approve. I rebased and squashed.

Sorry for this. I tried to split PR into multiple smaller PRs for easier review.

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Pull request merged**